### PR TITLE
fix(Field.Date): define empty value to trigger required error

### PR DESCRIPTION
**Pull request by joakbjerk**
proposal to fix https://github.com/dnbexperience/eufemia/issues/2970

### CODE REVIEW

**Comment by joakbjerk** [comment]
> Thanks for the great proposal 🚀 I tried to reproduce the same bug as reported in #2970, with this PR/fix in the following [CSB](https://codesandbox.io/p/sandbox/eufemia-starter-forked-vcctqq?file=%2Fsrc%2FApp.tsx%3A1%2C55). In the CSB I experienced the same bug/issue reported, selecting a date and then clicking tilbakestill(then no required error is displayed). That makes me wonder, what does this PR/fix improve in terms of displaying required error? 🤔
> 
> Anyhow, I'm not actually sure if what I've reported in #2970, as a bug actually is a bug? Or should the Field.Date actually behave different than other Field-components, because a datepicker is different? If we would want to improve this, perhaps the actual change needed would be in the DatePicker component, if so, I guess that would be a bit bigger job than a minor fix in Field.Date.
> 
> Many thoughts, perhaps we should discuss this with @tujoworker and the team

Ahh yeah, it seems to only work if you remove the date by pressing backspace, so this fix is not yet ready 🙏


Screenshot 2023-12-07 at 13 54 15

**Comment by langz** [comment]
> > Thanks for the great proposal 🚀 I tried to reproduce the same bug as reported in #2970, with this PR/fix in the following [CSB](https://codesandbox.io/p/sandbox/eufemia-starter-forked-vcctqq?file=%2Fsrc%2FApp.tsx%3A1%2C55). In the CSB I experienced the same bug/issue reported, selecting a date and then clicking tilbakestill(then no required error is displayed). That makes me wonder, what does this PR/fix improve in terms of displaying required error? 🤔
> > Anyhow, I'm not actually sure if what I've reported in #2970, as a bug actually is a bug? Or should the Field.Date actually behave different than other Field-components, because a datepicker is different? If we would want to improve this, perhaps the actual change needed would be in the DatePicker component, if so, I guess that would be a bit bigger job than a minor fix in Field.Date.
> > Many thoughts, perhaps we should discuss this with @tujoworker and the team
> 
> Ahh yeah, it seems to only work if you remove the date by pressing backspace, so this fix is not yet ready 🙏
> 
> Screenshot 2023-12-07 at 13 54 15

But that could also be the "correct way" or "only way" of handling displaying the required message for this component, as it could be that it doesn't make sense to display it when pressing "tilbakestill"-button and that potentially could be a heavy change in DatePicker 🤔

**Comment by joakbjerk** [comment]
> > > Thanks for the great proposal 🚀 I tried to reproduce the same bug as reported in #2970, with this PR/fix in the following [CSB](https://codesandbox.io/p/sandbox/eufemia-starter-forked-vcctqq?file=%2Fsrc%2FApp.tsx%3A1%2C55). In the CSB I experienced the same bug/issue reported, selecting a date and then clicking tilbakestill(then no required error is displayed). That makes me wonder, what does this PR/fix improve in terms of displaying required error? 🤔
> > > Anyhow, I'm not actually sure if what I've reported in #2970, as a bug actually is a bug? Or should the Field.Date actually behave different than other Field-components, because a datepicker is different? If we would want to improve this, perhaps the actual change needed would be in the DatePicker component, if so, I guess that would be a bit bigger job than a minor fix in Field.Date.
> > > Many thoughts, perhaps we should discuss this with @tujoworker and the team
> > 
> > 
> > Ahh yeah, it seems to only work if you remove the date by pressing backspace, so this fix is not yet ready 🙏
> > Screenshot 2023-12-07 at 13 54 15
> 
> But that could also be the "correct way" or "only way" of handling displaying the required message for this component, as it could be that it doesn't make sense to display it when pressing "tilbakestill"-button and that potentially could be a heavy change in DatePicker 🤔

Yeah we probably should discuss this in more detail

File: packages/dnb-eufemia/src/extensions/forms/Field/Date/__tests__/Date.test.tsx [hunk 1]
```diff
@@ -1,10 +1,76 @@
...
+      datepicker.querySelector('.dnb-form-status__text')
+    ).not.toBeInTheDocument()
+
+    act(() => {
+      inputs[inputs.length - 1].focus()
+      inputs[inputs.length - 1].setSelectionRange(4, 4)
+    })
+
+    await userEvent.keyboard(
+      '{Backspace}{Backspace}{Backspace}{Backspace}{Backspace}{Backspace}{Backspace}{Backspace}'
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Field/Date/__tests__/Date.test.tsx:34]:
```suggestion
      '{Backspace>8}'
```

**Review by langz — APPROVED** [review]

**Pull request merged**